cpp/src/arrow/dataset/file_csv.h

```diff
@@ -38,6 +38,13 @@ class ARROW_DS_EXPORT CsvFileFormat : public FileFormat {
  public:
   /// Options affecting the parsing of CSV files
   csv::ParseOptions parse_options = csv::ParseOptions::Defaults();
+  /// Number of header rows to skip (see arrow::csv::ReadOptions::skip_rows)
+  int32_t skip_rows = 0;
+  /// Column names for the target table (see arrow::csv::ReadOptions::column_names)
+  std::vector<std::string> column_names;
```

Comment:
I'm still -0 on including this option. It makes sense for a single file reader but I don't think datasets needs to provide two approaches for renaming columns.

By contrast, skip_rows and autogenerate are necessary here to accommodate the cases where files
- have non-csv front matter like a license comment which we shouldn't attempt to parse
- have data in their first row which we shouldn't interpret as column names

respectively.

@jorisvandenbossche @nealrichardson what do you think?

Reply:
I see what you're saying @bkietz, but to make the case for including it:

1. Maybe it's redundant, but is eliminating that redundancy worth adding surprises to our UI (where the usual options supported in the file reader are mostly supported in datasets, except for this one)?
2. Is it redundant? What is the other approach for renaming columns in a dataset? If it's ScannerBuilder::Project, that's arguably not renaming the dataset, just the scan results. It's also not trivial to rename columns this way if the reason why you're providing them is because there aren't any in the csv file. If it's that you can provide a Schema when you create the dataset, that means you also have to know/specify all of the column types, which you may not want to do or be able to do.

Reply:
AFAIK, `column_names` is not meant to replace existing columns (and thus do an implicit renaming), it's to specify column names if they are not present in the file. 

That way, I think it makes a lot of sense to include it here. If you have a set of regular CSV files (all having the same number of columns, in the same order, etc) that don't have names embedded, you would otherwise not be able to read those using the Datasets API.